experimental/AlgebraicStatistics/src/DiscreteGraphicalModels.jl

```diff
@@ -0,0 +1,86 @@
+function bron_kerbosch(G, R, P, X)
+
+    cliques = []
+
+    if length(P) == 0 && length(X) == 0
+        
+        push!(cliques, R)
+    end
+
+    Q = P
+    Y = X
+
+    for v in Q
+
+        new_R = vcat(R, v);
+        new_Q = [p for p in Q if p in neighbors(G, v)]
+        new_Y = [p for p in Y if p in neighbors(G, v)]
+
+        cliques = vcat(cliques, bron_kerbosch(G, new_R, new_Q, new_Y))
+
+        Q = filter(i -> i != v, P)
+        Y = vcat(X, v)
+    end
+
+    return cliques
+end
+
+function maximal_cliques(G::Graph{Undirected})
+
+    max_cliques = bron_kerbosch(G, [], vertices(G), []);
+
+    unique(sort.(max_cliques))
+end
+
+
+
+
+@doc raw"""
+    graphical_model(G::Graph{Undirected}, S::GaussianRing, k_var_name::String="k")
+
+A parametric statistical model associated to an undirected graph.
+It contains an undirected graph `G`, a GaussianRing `S` where the vanishing ideal of the model naturally lives, 
+and a parameter ring whose variables `k[i,j]` correspond to the edges `i - j` in `G`. 
+
+## Examples
+
+``` jldoctest undirected_ggm
+julia> M = graphical_model(graph_from_edges([[1,2], [2,3]]), gaussian_ring(3))
+Gaussian graphical model on an undirected graph with edges:
+(1, 2), (2, 3)
+```
+"""
+function graphical_model(G::Graph{Undirected}, S::MarkovRing, t_var_name::String="t")
```

Comment:
Here and elsewhere where `BLA_var_name` occurs: I recommend replacing using our `VarName` type, which makes it more flexible (also symbols and chars can be passed in) while also making it possible drop the `_var_name` suffix which is rather unusual:
```suggestion
function graphical_model(G::Graph{Undirected}, S::MarkovRing, t::VarName="t")
```
or
```suggestion
function graphical_model(G::Graph{Undirected}, S::MarkovRing, t::VarName=:t)
```